python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -71,6 +75,39 @@ def __call__(self, df):
         return magDiff
 
 
+class ExtinctionCorrectedMagDiff(DataFrameAction):
+    magDiff = ConfigurableActionField(doc="Action that returns a difference in magnitudes",
+                                          default=MagDiff, dtype=DataFrameAction)
+    ebv = Field(doc="E(B-V) Column", dtype=str, default="ebv")
+    extinctionCoeffs = DictField(
+        doc="Dictionary of extinction coefficients for conversion from E(B-V) to extinction, A_filter."
+            "Key must be the band",
+        keytype=str, itemtype=float, optional=True,
+        default=None)
+
+    @property
+    def columns(self):
+        return self.magDiff.columns + (self.ebv,)
+
+    def __call__(self, df):
+        diff = self.magDiff(df)
+        if not self.extinctionCoeffs:
+            _LOG.warn("No extinction Coefficients. Not applying extinction correction")
+            return diff
+
+        ebv = df[self.ebv].values
+        col1Band = self.magDiff.col1.split('_')[0]
+        col2Band = self.magDiff.col2.split('_')[0]
+        av1 = self.extinctionCoeffs[col1Band]
+        av2 = self.extinctionCoeffs[col2Band]
+        correction = (av1 - av2) * ebv
+
+        if self.magDiff.returnMillimags:
+            correction = correction*1000.0
+
+        return diff - correction*u.mag
```

Comment:
Upon further review, this is incorrect here and elsewhere (everywhere?) in `analysis_drp`, I fear.  The units should not be `u.mag` but `u.mmag` if we're using millimags.  This should be a separate ticket (@sr525 ).

Reply:
No prob. I changed MagDiff and this Action. I went to give your `ColorDiff` the same treatment but found that it was just a numpy array, so I'll be leaving the 1000* there. 
```    color_diff = color_diff.to(u.mmag)
AttributeError: 'numpy.ndarray' object has no attribute 'to'
```